bindings/py/tests/sparse_matrix_test.py

```diff
@@ -162,6 +162,70 @@ def test_setAllNonZeros(self):
         error('setAllNonZeros, out of order')
 
 
+  def test_nNonZerosPerRow_allRows(self):
+    for name, denseMatrix, expected in (
+        ("Test 1",
+         numpy.array([[1, 1, 0, 0, 1, 1],
+                      [0, 0, 1, 1, 0, 0],
+                      [0, 0, 1, 0, 0, 1],
+                      [1, 0, 1, 1, 0, 0],
+                      [0, 0, 0, 0, 0, 1],
+                      [0, 0, 0, 0, 0, 0],
+                      [1, 1, 1, 1, 1, 1],
+                      [0, 0, 1, 1, 0, 1]], dtype="float32"),
+         [4, 2, 2, 3, 1, 0, 6, 3]),
+        ("No rows",
+         numpy.zeros((0,0), dtype="float32"),
+         []),
+        ("Empty rows",
+         numpy.array([[],
+                      [],
+                      []], dtype="float32"),
+         [0, 0, 0]),
+    ):
+      m = SparseMatrix(denseMatrix)
+
+      numpy.testing.assert_equal(m.nNonZerosPerRow(),
```

Comment:
Is this unittest-aware? If not, then it's probably better to do `self.assertTrue(numpy.array_equal(...))`